test/mix/tasks/ecto_test.exs

```diff
@@ -0,0 +1,24 @@
+defmodule Mix.Tasks.EctoTest do
+  use ExUnit.Case, async: true
+
+  test "provide help" do
+    Mix.Tasks.Ecto.run []
+    version = "Ecto v#{Ecto.version}"
+    assert_received {:mix_shell, :info, [version]}
```

Comment:
You're missing a `^` to match on the `version` variable, that's where your warning comes from. :)